### DISCUSSION TITLE
Is JIT compilation only implemented outside of distributed execution?

### DISCUSSION THREAD

**Discussion by DaveParr** [Nixtla/statsforecast#665] [Q&A]
This is not particularly rigorous benchmarmking, but I think I've observed something that AFAIK is undocumented?

When running a cross-validation -> forecasting set of operations with 3 models and 5 windows I'm observing that without a dask distributed backend the first run is about 20 seconds, and the subsequent runs are about 2 seconds. However, when I run the same code through a dask engine that every run is 30 seconds. 

I understand why the overhead might exist, but I was also surprised that subsequent runs in distributed mode have the same execution time of 30 seconds, not less. Is this expected, or have I done something wrong?

My inference from this is that the JIT compilation step is skipped (or not cached) when running in fugue/dask. Is that true?

**Comment by jmoralez** [comment]
Hey. Are you able to see the dask dashboard? I'm not sure but I think one task is created per serie, so the overhead of that can be very big. If that's the case I think you could open an issue requesting more control over the partitioning. If you have 4 workers for example and 100 series, you could ask for 4 partitions and each worker would process one task (of approximately 25 series). I think now 100 tasks are created, which can introduce a lot of overhead.

**Reply by DaveParr** [reply]
In my current project the dashboard isn't accessible. I get this error when I head to the dashboard URL in the log:

> This site can’t be reached 127.0.0.1 refused to connect.

It's a django app, running in docker, so that's probably why there are connection issues. The log explicitly states:

>  dashboard at:  http://127.0.0.1:33607/status

Likely because of this exact scenario.

I suspect the answer is to make a test project on my side that is less complicated and see what that says, unless you have any tips for hos to make the dashboard more accessible?

**Reply by jmoralez** [reply]
You can fix the dashboard port and then bind it to the container to access it from the outside but it's probably not worth it. I'll try to investigate it on my side. Which models are you using?

**Reply by DaveParr** [reply]
``` python
models = [
        AutoARIMA(season_length=seasonality, approximation=True),
        DynamicOptimizedTheta(season_length=seasonality),
    ]
```
